### PR TITLE
test: refactor exec tests to AVA

### PR DESCRIPTION
**Pull request by nfischer**
This is an automated PR, so double check that I got all the test cases. I'll follow up once I've verified that this translates everything correctly.

### CODE REVIEW

File: ava-test/exec.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    t.is(stderr2, '');
+    t.end();
+  });
+});
+
+test.cb('callback as 3rd argument (silent:true)', t => {
+  shell.exec(`${JSON.stringify(process.execPath)} -e 'console.log(5678);'`, { silent: true }, (code3, stdout3, stderr3) => {
+    t.is(code3, 0);
+    t.is(stdout3, '5678\n');
+    t.is(stderr3, '');
```
Comment (freitagbr) [inline ava-test/exec.js]: This and the test above can use `code`, `stdout`, and `stderr` and callback arguments.

Comment (nfischer) [inline ava-test/exec.js]: Done

File: ava-test/exec.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+//
+// Valids
+//
+
+//
+// sync
+//
+
+test('check if stdout goes to output', t => {
+  const result = shell.exec(`${JSON.stringify(process.execPath)} -e 'console.log(1234);'`);
```
Comment (freitagbr) [inline ava-test/exec.js]: Do the single-quotes work on Windows?

Comment (nfischer) [inline ava-test/exec.js]: They might not. I'll change them back and see

**Comment by freitagbr** [comment]
LGTM

**Pull request merged**